src/Centreon/Infrastructure/Monitoring/TimelineRepositoryRDB.php

```diff
@@ -0,0 +1,430 @@
+        NULL AS `sticky`,
+        NULL AS `notify_contacts` 
+        FROM `:dbstg`.`logs` l
+        WHERE l.host_id = :hostId AND l.service_id = :serviceId";
+
+        $collector->addValue(":hostId", $hostId);
+        $collector->addValue(":serviceId", (int) $serviceId);
+
+        // set ACL limitations
+        if (!$this->isAdmin()) {
+            $sql .= " INNER JOIN `:dbstg`.`centreon_acl` AS service_acl ON service_acl.host_id = s.host_id
+                  AND service_acl.service_id = s.service_id
+                  AND service_acl.group_id IN (" . $this->accessGroupIdToString($this->accessGroups) . ")";
+        }
+
+        //Group to avoid duplicate entries
+        $sql .= ' GROUP BY l.log_id';
+
+        return $sql;
+    }
+
+    /**
+     * Generate SQL query for comments
+     *
+     * @param \Centreon\Infrastructure\CentreonLegacyDB\StatementCollector $collector
+     * @param int $hostId
+     * @param int $serviceId
+     * @return string
+     */
+    protected function generateCommentsQuery(StatementCollector $collector, int $hostId, int $serviceId = null): string
+    {
+        $sql = "SELECT
+		CONCAT('C', c.comment_id) AS `eventId`,
+        '". CommentEventObject::EVENTTYPE ."' AS `eventType`,
+        c.comment_id AS `id`,
+		c.data AS `output`,
+		c.entry_time AS `timestamp`,
+        NULL AS `status`,
+        c.type AS `type`,
+        NULL AS `retry`,
+        c.author AS `contact`,
+        NULL AS `command`,
+        c.persistent AS `persistent`,
+        NULL as `deletion_time`,
+        NULL AS `start_time`,
+        NULL AS `end_time`,
+        NULL AS `actual_start_time`,
+        NULL AS `actual_end_time`,
+        NULL AS `duration`,
+        NULL AS `started`,
+        NULL AS `cancelled`,
+        NULL AS `fixed`,
+        NULL AS `sticky`,
+        NULL AS `notify_contacts`
+        FROM `:dbstg`.`comments` c
+        WHERE c.host_id = :hostId AND c.service_id = :serviceId";
+
+        $collector->addValue(":hostId", $hostId);
+        $collector->addValue(":serviceId", (int) $serviceId);
+
+        // set ACL limitations
+        if (!$this->isAdmin()) {
+            $sql .= " INNER JOIN `:dbstg`.`centreon_acl` AS service_acl ON service_acl.host_id = s.host_id
+                  AND service_acl.service_id = s.service_id
+                  AND service_acl.group_id IN (" . $this->accessGroupIdToString($this->accessGroups) . ")";
+        }
+
+        //Group to avoid duplicate entries
+        $sql .= ' GROUP BY c.comment_id';
+
+        return $sql;
+    }
+
+    /**
+     * Generate SQL query for downtimes
+     *
+     * @param \Centreon\Infrastructure\CentreonLegacyDB\StatementCollector $collector
+     * @param int $hostId
+     * @param int $serviceId
+     * @return string
+     */
+    protected function generateDowntimesQuery(StatementCollector $collector, int $hostId, int $serviceId = null): string
+    {
+        $sql = "SELECT
+		CONCAT('D', d.downtime_id) AS `eventId`,
```

Comment:
IMHO, eventId should be an integer from only eventId
The unique id will be the association between eventType and eventId